firebase/src/core/data/gateways/env-gateway.ts

```diff
@@ -20,7 +20,7 @@ export class EnvGateway {
 
   /** Returns `true` if running in local environment, i.e, under Firebase Emulators. */
   get isLocalDevelopment(): boolean {
-    return process.env["FUNCTIONS_EMULATOR"] === "true";
+    return process.env["FUNCTIONS_EMULATOR"] === "true" || process.env["ENVIRONMENT"] === "DEV";
```

Comment:
When debugging scripts, even if the Firebase Emulator is running, the env `FUNCTIONS_EMULATOR` is false because we need to run the script in a different process (i.e, a different terminal window). To solve this issue, I suggest to add another env variable `ENVIRONMENT`, which must be set when running scripts and other executions that don't belong to firebase emulators itself

Reply:
You're right, this `ENVIRONMENT` doesn't make sense. I'm thinking in another possibility to "protect" the scripts from being run in production when they should be run locally.